tsktsk/db.py

```diff
@@ -0,0 +1,44 @@
+import sqlite3
+import threading
+from pathlib import Path
+from typing import Any, Iterable
+
+from pkg_resources import resource_filename
+from yoyo import get_backend, read_migrations
+
+
+class Database:
+    def __init__(self, db_path):
+        self.connection = sqlite3.connect(db_path)
+        self.connection.row_factory = sqlite3.Row
+
+    def execute(self, sql: str, params: Iterable[Any] = ()) -> None:
+        self.connection.execute(sql, params)
+
+    def fetchone(self, sql: str, params: Iterable[Any] = ()) -> Any:
+        return self.connection.execute(sql, params).fetchone()
+
+    def __enter__(self) -> "Database":
+        self.connection.__enter__()
+        return self
+
+    def __exit__(self, *args, **kwargs):
+        self.connection.__exit__(*args, **kwargs)
+
+
+namespace = threading.local()
```

Comment:
What is the thought behind linking this to a thread vs say the click context?

Reply:
Sqlite has some issues with sharing a connection between threads, and so python's sqlite lib checks for the current thread and throws an exception if a connection is used in another thread. I think the only part rn that's switching threads is the auth command. There is the option of disabling thread check when creating a connection, but I am not sure of the implications. I guess all of this can go away once we have connection pooling.